src/arkreactor/VM/State.cpp

```diff
@@ -232,7 +232,7 @@ namespace Ark
             std::string str_version = std::to_string(major) + "." +
                 std::to_string(minor) + "." +
                 std::to_string(patch);
-            throwStateError("Compiler and VM versions don't match: " + str_version + " and " + ARK_VERSION_STR);
+            throwStateError("Compiler and VM versions don't match: " + str_version + " and " + std::string(ARK_VERSION_STR, sizeof(ARK_VERSION_STR)));
```

Comment:
This is definitely something we had to fix, good catch

Though, may I suggest to directly update the `include/Ark/Constants.hpp.in` so that the null terminator is added to `ARK_VERSION_STR`, and we won't have to do a cast to `std::string` everywhere we want to use it?